velox/functions/sparksql/SIMDJsonFunctions.h

```diff
@@ -0,0 +1,182 @@
+            int64_t numberResult;
+            error = rawResult.get_int64().get(numberResult);
+            if (!error) {
+              ss << numberResult;
+              result.append(ss.str());
+            }
+            return error;
+          }
+          case ondemand::number_type::floating_point_number: {
+            double numberResult;
+            error = rawResult.get_double().get(numberResult);
+            if (!error) {
+              ss << numberResult;
+              result.append(ss.str());
+            }
+            return error;
+          }
+          default:
+            VELOX_UNREACHABLE();
+        }
+      }
+      case ondemand::json_type::boolean: {
+        bool boolResult;
+        error = rawResult.get_bool().get(boolResult);
+        if (!error) {
+          result.append(boolResult ? "true" : "false");
+        }
+        return error;
+      }
+      case ondemand::json_type::string: {
+        std::string_view stringResult;
+        error = rawResult.get_string().get(stringResult);
+        result.append(stringResult);
+        return error;
+      }
+      case ondemand::json_type::object: {
+        // For nested case, e.g., for "{"my": {"hello": 10}}", "$.my" will
+        // return an object type.
+        ss << rawResult;
+        result.append(ss.str());
+        return SUCCESS;
+      }
+      case ondemand::json_type::array: {
+        ss << rawResult;
+        result.append(ss.str());
+        return SUCCESS;
+      }
+      default: {
+        return UNSUPPORTED_ARCHITECTURE;
+      }
+    }
+  }
+
+  // This is a simple validation by checking whether the obtained result is
+  // followed by valid char. Because ondemand parsing we are using ignores json
+  // format validation for characters following the current parsing position.
+  bool isValidEndingCharacter(const char* currentPos) {
+    char endingChar = *currentPos;
+    if (endingChar == ',' || endingChar == '}' || endingChar == ']') {
+      return true;
+    }
+    if (endingChar == ' ' || endingChar == '\r' || endingChar == '\n' ||
+        endingChar == '\t') {
+      // These chars can be prior to a valid ending char.
+      return isValidEndingCharacter(currentPos++);
+    }
+    return false;
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& json,
+      const arg_type<Varchar>& jsonPath) {
+    ParserContext ctx(json.data(), json.size());
+    try {
+      ctx.parseDocument();
+      simdjson_result<ondemand::value> rawResult =
+          formattedJsonPath_.has_value()
+          ? ctx.jsonDoc.at_pointer(formattedJsonPath_.value().data())
+          : ctx.jsonDoc.at_pointer(getFormattedJsonPath(jsonPath).data());
+      // Field not found.
+      if (rawResult.error() == NO_SUCH_FIELD) {
+        return false;
+      }
+      auto error = extractStringResult(rawResult, result);
+      if (error) {
+        return false;
+      }
+    } catch (simdjson_error& e) {
```

Comment:
We should not have any exception here, every error should be represented in the return code

Reply:
Thanks for your comment! 
I found simdjson lib can throw exceptions for some cases.
For example, if we don't keep try/catch here, the below test will fail with an exception thrown:
"_The JSON document has an improper structure: missing or superfluous commas, braces, missing key_"
```c++
EXPECT_EQ(getJsonObject(R"({"hello"-3.5})", "$.hello"), std::nullopt);
```

Reply:
@Yuhta Jimmy, I wonder if that's why we are seeing unhandled "Unexpected trailing content in the JSON input" errors in some queries: T175957555 . Since throwing exceptions is expensive (under TRY), I wonder if there is a way to tell simdjson to not throw but rather return error code. Maybe we can open an issue in simdjson GitHub repo and ask about that.

Reply:
@PHILO-HE @mbasmanova Certain simdjson functions are throwing, you can tell from the signature, but all throwing functions have a corresponding version that is returning error code.  So we need to find out which function is throwing and replace them, then we can remove the catch here.